src/states/common/base.rs

```diff
@@ -150,13 +150,13 @@ pub trait Base: Display {
             UnsentUserMessage {
                 peer_addr,
                 msg,
-                msg_id,
-            } => self.handle_unsent_message(peer_addr, msg, msg_id, outbox),
+                token,
+            } => self.handle_unsent_message(peer_addr, msg, token, outbox),
```

Comment:
Maybe the functions `handle_unsent_message` and `handle_sent_message` should have the argument name (and type) changed in their signature, too.